spec/models/collection_spec.rb

```diff
@@ -1,5 +1,82 @@
 require 'rails_helper'
 
 RSpec.describe Collection do
-  # "Add your tests here"
+  let(:collection_type) { double(:brandable, true) }
+  let(:collection_type_gid) { FactoryBot.create(:user_collection_type).gid }
+  let(:collection) { Collection.create(title: ['Test collection'],
+                                       collection_type_gid: collection_type_gid,
+                                       visibility: Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PUBLIC) }
+  let(:fs1) { FactoryBot.create(:file_set) }
+  let(:fs2) { FactoryBot.create(:file_set) }
+  let(:files) { [[fs1.title, fs1.id], [fs2.title, fs2.id]] }
+
+
+  describe "#default_thumbnail_id", :clean do
+    context "with no thumbnails available" do
+
+      before do
+        allow_any_instance_of(Hyrax::Forms::CollectionForm).to \
+          receive(:send).with(:all_files_with_access).and_return([])
+      end
+      it "returns nil" do
+        expect(collection.default_thumbnail_id).to be_nil
+      end
+    end
+    context "with no thumbnails available" do
```

Comment:
```suggestion
    context "with thumbnails available" do
```

Reply:
@dlpierce re-pushed fixing this and with some refactoring, changing the before_save callback to a before_update